### PR TITLE
Fixed json tags for volumes

### LINKED ISSUE #8755: Pod spec.volumes.rbd included when empty
"rbd:null" is included in a `kubectl get pods -ojson` (during the e2e tests on tmpfs).  I don't know whether the root cause is that the pointer is being populated or whether we just need omitEmpty...

```
...
 "volumes": [
                    {
                        "name": "test-volume",
                        "emptyDir": {
                            "medium": "Memory"
                        },
                        "rbd": null
                    }
                ],
...
```

### PR DESCRIPTION
**Pull request by markturansky**
Several "omitempty" tags missing.

Resolves #8755.

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ type VolumeSource struct {
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
 	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
-	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim"`
```
Comment (bgrant0607) [inline pkg/api/types.go]: omitempty implies optional. Really we shouldn't have any json tags in this file, but so long as we do, we might as well make the tags correct and consistent. All volume sources should have omitempty here, or just remove the json tags entirely.

**Comment by pmorie** [comment]
Fyi @rootfs

File: pkg/api/v1beta3/types.go [hunk 2]
```diff
@@ -252,7 +252,7 @@ type PersistentVolumeSource struct {
 	// NFS represents an NFS mount on the host
 	NFS *NFSVolumeSource `json:"nfs,omitempty" description:"NFS volume resource provisioned by an admin"`
 	// RBD represents a Rados Block Device mount on the host that shares a pod's lifetime
-	RBD *RBDVolumeSource `json:"rbd" description:"rados block volume that will be mounted on the host machine"`
+	RBD *RBDVolumeSource `json:"rbd,omitempty" description:"rados block volume that will be mounted on the host machine"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: What about v1beta1 and v1beta2? We should keep them up to date until we remove them.

**Comment by bgrant0607** [comment]
Thanks, @markturansky.

File: pkg/api/v1beta3/types.go (hunk 2 above)
Comment (markturansky) [inline pkg/api/v1beta3/types.go]: I noticed they were one way while v1beta3 was another.  I thought maybe it was purposeful.  I believe they used to be there.

I'll add 'omitempty' to the other versions, too.

File: pkg/api/types.go (hunk 1 above)
Comment (markturansky) [inline pkg/api/types.go]: I'll add omitempty to each.  I believe Tim tried removing all the json tags and failed.

**Comment by markturansky** [comment]
@bgrant0607 I added 'omitempty' to internal types and v1beta1/2, per your request.  

Also, I removed 'omitempty' from PersistentVolumeClaimVolumeSource.ClaimName.  Validation already requires the name, so omitempty makes no sense.

cc @nikhiljindal this continues #8338

File: pkg/api/v1beta3/types.go (hunk 2 above)
Comment (markturansky) [inline pkg/api/v1beta3/types.go]: nm, i see these were all added by Nikhil but only to v1 and v1beta3. They are all the same now.

**Comment by thockin** [comment]
LGTM

**Comment by bgrant0607** [comment]
Thanks. LGTM.

**Comment by thockin** [comment]
needs rebase

**Comment by markturansky** [comment]
Rebased and added 'omitempty' to the couple new volumes that were added to PV.  Waiting on green.

**Comment by markturansky** [comment]
@thockin Travis returned green.

**Pull request merged**